### PR TITLE
Switch to cmake.

### PR DESCRIPTION
**Pull request by Denaun**

### CODE REVIEW

**Comment by cristiano-santoni** [comment]
Looks good to me.
In the future however I'd like to enforce out of source build and testing (so the ignore on cmake files, makefile and test directory would be removed).

**Comment by Denaun** [comment]
You have to approve it!

You mean having the makefiles in the repo?
Personally I would avoid that, keeping just cmake allows using whatever the user wants (Make, Ninja, VS, …)

**Comment by cristiano-santoni** [comment]
I know I have to approve it, I just wanted to have a conversation :'(
I meant, as LLVM does, that if you specify a directory that is within the source base it won't build.
Makefile should not be in the repo and it will never be generated in there by cmake so it will be possible to remove the rule in the .ignore

**Comment by Denaun** [comment]
Oh ok, I didn't get that.
Yes, I agree on forcing not to use the base folder.
I think the gitignore is still useful, since CMake generates stuff anyway (like CMakeCache)

(You closed without approving! 😃)

**Comment by cristiano-santoni** [comment]
Cmake generate the cache files in the place were you build it, if you are not within the source base, those files will be generate somewhere else, like in a build directory at the same level as the repo directory:
mkdir build
cd build
cmake ../fistz

**Comment by Denaun** [comment]
That's true, my bad.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.6)
```
Comment (cristiano-santoni) [inline CMakeLists.txt:1]: maybe we could try to import this from llvm cmake project to be use to be in sync

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,14 @@
...
+# CMake
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
+cmake_install.cmake
+install_manifest.txt
+compile_commands.json
+CTestTestfile.cmake
```
Comment (cristiano-santoni) [inline .gitignore:40]: to be removed when we'll enforce out of source build

File: CMakeLists.txt (hunk 1 above)
Comment (Denaun) [reply CMakeLists.txt:1]: Yep, I just took the one used by travis.
Looks like LLVM uses 3.4.3

**Review by cristiano-santoni — APPROVED** [review]

**Pull request merged**